pulley/src/interp.rs

```diff
@@ -1619,31 +1650,41 @@ impl OpVisitor for Interpreter<'_> {
                 self.state[operands.dst].set_u64(result);
                 ControlFlow::Continue(())
             }
-            None => self.done_trap::<crate::XDiv64U>(),
+            None => self.done_trap_kind::<crate::XDiv64U>(Some(TrapKind::DivideByZero)),
         }
     }
 
     fn xrem32_s(&mut self, operands: BinaryOperands<XReg>) -> ControlFlow<Done> {
         let a = self.state[operands.src1].get_i32();
         let b = self.state[operands.src2].get_i32();
-        match a.checked_rem(b) {
+        let result = if a == i32::MIN && b == -1 {
```

Comment:
I'll note that this is an interesting case of a divergence in Rust semantics and WebAssembly semantics.